scripts/update_stm32_package.py

```diff
@@ -132,6 +129,33 @@ def update_cubes():
         logging.info("%s", "Add option --noclean")
         args.noclean = True
 
+# prevent losing data in an unclean repo
+git_status = subprocess.check_output(
+    ("git", "status", "-s"),
+    cwd=SCRIPT_DIR,
+).splitlines()
+# status = [x.decode("utf-8") for x in git_status]
+for status in git_status:
+    if not status.decode("utf-8").startswith("?") and not status.decode(
+        "utf-8"
+    ).startswith("!"):
+        logging.error(
+            "%s",
+            "It seems that script repo "
+            + str(SCRIPT_DIR)
+            + "\n           is not clean:\n           git status -s:"
+            + status.decode("utf-8"),
+        )
+        logging.error("%s", "It is suggested to clean your repo first.")
+
+        print("Do you want to continue update ?")
+        res = input("(Enter y/n) ").lower()
+        while res not in ("y", "n"):
+            res = input("(Enter y/n) ").lower()
+        if res == "n":
+            sys.exit()
+        break
```

Comment:
Ooops, I missed this one. I will implement it.

Reply:
This looks interesting, but finally, I wonder whether this is a good idea, because pygit2 is not part of zephyr requirements.
So adding this dependencies seems too much only for that (we are currently very few persons to update stm32 cube), especially knowing that I already have a working solution (even if more complex).